src/pages/LoginPage.jsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Container,
+  Paper,
+  TextField,
+  Button,
+  Typography,
+  FormLabel,
+  IconButton,
+  InputAdornment,
+} from '@mui/material';
+import { Visibility, VisibilityOff } from '@mui/icons-material';
+import { login } from '../auth/authThunks';
+import { selectError, selectAuthToken } from '../auth/authSelectors';
+import { clearError } from '../auth/authSlice';
+import ToastMessage from './ToastMessage.jsx';
+import { useNavigate } from 'react-router-dom';
+
+const LoginPage = () => {
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const error = useSelector(selectError);
+  const authToken = useSelector(selectAuthToken);
+
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [showPassword, setShowPassword] = useState(false);
+  const [toast, setToast] = useState(null);
+
+  const handleLogin = async (event) => {
+    event.preventDefault();
+
+    await dispatch(login({ username, password }));
+
+    if (authToken) {
+      navigate('/home');
+    } else {
+      setToast(
+        <ToastMessage type="error" message="Invalid username or password." />
+      );
+    }
+  };
+
+  const handleTogglePasswordVisibility = () => {
+    setShowPassword(!showPassword);
+  };
+
+  useEffect(() => {
+    if (error) {
+      setToast(<ToastMessage type="error" message={error} />);
+      dispatch(clearError());
+    }
+  }, [error, dispatch]);
+
```

Comment:
Add a logic here to redirect the user to the **dashboard home** when the authToken is present.